src/kaleidoscope/key_defs.h

```diff
@@ -202,6 +203,45 @@ typedef kaleidoscope::Key Key_;
 #define ___ Key_Transparent
 #define XXX Key_NoKey
 
+namespace kaleidoscope {
+constexpr bool isKeyboardKey(Key key) {
+  return ((key.getFlags() & (RESERVED | SYNTHETIC)) == 0);
+}
+constexpr bool isKeyboardModifier(Key key) {
+  return (isKeyboardKey(key) &&
+          (key.getKeyCode() >= HID_KEYBOARD_FIRST_MODIFIER &&
+           key.getKeyCode() <= HID_KEYBOARD_LAST_MODIFIER));
+}
+constexpr bool isKeyboardShift(Key key) {
+  return (isKeyboardModifier(key) &&
+          ((key.getKeyCode() == HID_KEYBOARD_LEFT_SHIFT ||
+            key.getKeyCode() == HID_KEYBOARD_RIGHT_SHIFT) ||
+           ((key.getFlags() & SHIFT_HELD) != 0)));
+}
+
+constexpr bool isLayerKey(Key key) {
+  return (key.getFlags() == (SYNTHETIC | SWITCH_TO_KEYMAP));
+}
+constexpr bool isLayerShift(Key key) {
+  return (isLayerKey(key) &&
+          (key.getKeyCode() >= LAYER_SHIFT_OFFSET));
+}
+
+constexpr uint8_t hid_type_flags_mask = HID_TYPE_MASK;
```

Comment:
Are these intended as private? If so, they should get named_ to make that clear.

Reply:
I didn't give any thought to privacy here. I did consider using a namespace (e.g. `kaleidoscope::key_flags::hid_type_mask`), which I think would be a good idea, but I don't see any reason to make these constants private (nor do I object to it, for that matter).